File: app/src/main/java/com/shevelev/comics_viewer/activities/main/ComicsSortInfo.java
package com.shevelev.comics_viewer.activities.main;

import com.shevelev.comics_viewer.dal.dto.Comics;

import java.util.Comparator;

/**
 * Data for comics sort
 */
public class ComicsSortInfo
{
    private Comparator<Comics> comparator;

    /** Need revers order     */
    private boolean reverse;

    public ComicsSortInfo(Comparator<Comics> comparator, boolean reverse)
    {
        this.comparator = comparator;
        this.reverse = reverse;
    }

    public Comparator<Comics> getComparator()
    {
        return comparator;
    }

    public boolean isReverse()
    {
        return reverse;
    }
}

File: app/src/main/java/com/shevelev/comics_viewer/activities/main/ComicsSortInfo.kt
package com.shevelev.comics_viewer.activities.main

import com.shevelev.comics_viewer.dal.dto.Comics
import java.util.*

/**
 * Data for comics sort
 */
class ComicsSortInfo(val comparator: Comparator<Comics>,
                     /** Need revers order      */
                     val isReverse: Boolean)